### PR TITLE
Variant carousel component

### LINKED ISSUE #15895: Variant carousel component
Provide component. Check if we can re-use product-carousel.component (otherwise use carousel.component).

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

**Review by ChristophHi — CHANGES_REQUESTED** [review]
We said we wanted to introduce a configuration that controls the variant search availability and also needs to be taken into account for the component.
Let me do this in the facade branch first, then I will let you know

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]
seems like that for example this commit https://github.com/SAP/spartacus/pull/15911/commits/aea775ec45f41d57a695dd9e34ddd9f683ba2f2e is in the feature branch, but not yet in the epic?

i guess it would make sense to update the epic from develop, so that we only squash merge the 'feature'

**Comment by Uli-Tiger** [comment]
@Larisa-Staroverova
`Merge branch 'develop' into feature/](https://github.com/SAP/spartacus/pull/15911/commits/d665db1d0cce6e65612fa9d3b848ed58b57fb3f0)https://github.com/SAP/spartacus/issues/15895`
Please always merge develop -> epic -> feature!

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        },
+        {
+          provide: ConfiguratorCommonsService,
+          useClass: MockConfiguratorCommonsService,
+        },
+      ],
+    }).compileComponents();
+  });
+
+  it('should create a component without variants', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts:161]: This test is failing for me

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+} from '@spartacus/product-configurator/common';
+import {
+  Configurator,
+  ConfiguratorCommonsService,
+} from '@spartacus/product-configurator/rulebased';
+import { ConfiguratorTestUtils } from '../../testing/configurator-test-utils';
+import { Component, Input, Pipe, PipeTransform } from '@angular/core';
+import { CommonConfiguratorTestUtilsService } from '../../../common/testing/common-configurator-test-utils.service';
+
+const PRODUCT_DESCRIPTION = 'Here is a product description';
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts:21]: Let's have the formatting of constants unified. Some are upper case, others not. I would rather do them in camel case style

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      map((variants) => {
+        return variants.map((variant) =>
+          this.productService.get(variant.productCode)
+        );
+      })
+    );
+
+  constructor(
+    protected productService: ProductService,
+    protected translation: TranslationService,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.ts]:
```suggestion
    protected translationService: TranslationService,
```

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    protected configRouterExtractorService: ConfiguratorRouterExtractorService,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.ts]:
```suggestion
    protected configuratorRouterExtractorService: ConfiguratorRouterExtractorService,
```

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/core/state/reducers/configurator.reducer.spec.ts [hunk 5]
```diff
@@ -352,22 +352,20 @@ describe('Configurator reducer', () => {
...
+      ];
+      const action = new ConfiguratorActions.SearchVariantsSuccess({
         ownerKey: configuration.owner.key,
-        overview: overview,
+        variants: variants,
       });
       const state = StateReduce.configuratorReducer(undefined, action);
 
-      expect(state.overview).toEqual(overview);
-      expect(state.priceSummary).toBe(priceSummary);
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/core/state/reducers/configurator.reducer.spec.ts:370]: did you intend to delete this existing test?

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+      htmlElem,
+      '.cx-variant-carousel-container'
+    );
+
+    CommonConfiguratorTestUtilsService.expectElementNotPresent(
+      expect,
+      htmlElem,
+      'cx-carousel'
+    );
+  });
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts:175]: test fails if i execute it locally
```Expected 1 to be 0, 'expected element identified by selector 'cx-carousel' to be NOT present, but it is!```

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]

**Review by ChristophHi — APPROVED** [review]

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts:161]: Now it works

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts:21]: Done!

File: feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/variant-carousel/configurator-variant-carousel.component.spec.ts:175]: Now it works

**Review by Uli-Tiger — APPROVED** [review]

**Comment by ChristophHi** [comment]
Closed as it was merged as part of another PR

**Pull request closed without merging**